### PR TITLE
chore: add governance

### PR DESCRIPTION
**Pull request by ematipico**
## Summary

A first draft of the governance. All @biomejs/core-contributors must approve this PR before being merged.

## Test Plan

Review it and approve it

### CODE REVIEW

**Review by unvalley — APPROVED** [review]

File: GOVERNANCE.md
Comment (nissy-dev) [inline GOVERNANCE.md:76]: Maybe, forgot to delete it.

```suggestion

```

**Review by nissy-dev — APPROVED** [review]
LGTM :+1: Thank you for spending much time on thinking the governance model!

**Comment by unvalley** [comment]
It might be okay to add this later, but it would be good to have an example of what specific contributions should be made in order to be in a certain role. I haven't seen that in many other repositories.

File: GOVERNANCE.md
Comment (denbezrukov) [inline GOVERNANCE.md:5]: Are there specific criteria for someone to be considered a "Core Contributor" or "Maintainer"?
Are there any conditions under which a Lead might lose their status or privileges?

File: GOVERNANCE.md
Comment (denbezrukov) [inline GOVERNANCE.md:40]: Are there guidelines or criteria for what constitutes a "significant explanation" when nominating someone?

**Review by denbezrukov — APPROVED** [review]

File: GOVERNANCE.md
Comment (bnb) [inline GOVERNANCE.md:100]: One of the hardest lessons I learned from the io.js Governance Structure that ended up as the Node.js Governance Structure was that the decision to _never_ remove inactive members was perhaps one of the biggest barriers to progress. Eventually, we introduced an emeritus system that IMO has been extremely helpful in distinguishing who is an active contributor vs. who once was but is not longer active.

The Emeritus system also helps ensure a more secure environment, since inactive accounts don't have push permissions, while still allowing for recognition and an easy way to start working again (simply ask to be re-added to the project from emeritus status, and you will be).

To be in good conscience, I feel obligated to at least leave a comment urging you to consider a way to build-in a system to step-down or off-board people who are inactive after having worked in such a system without an escape hatch like Emeritus for so long.

File: GOVERNANCE.md
Comment (bnb) [inline GOVERNANCE.md:105]: As a non-binary person, I appreciate the last part here, but generally disclosing gender isn't something I'd expect to be done. My _guess_ is that you meant pronouns and used gender as a proxy for that, so here's two suggestions on how to tweak this:

```suggestion
Members are listed in alphabetical order. Members are free to use the full name, GitHub handle, or any other nickname they wish to be addressed. Members are free to disclose their gender and pronouns.
```

```suggestion
Members are listed in alphabetical order. Members are free to use the full name, GitHub handle, or any other nickname they wish to be addressed. Members are free to disclose their pronouns.
```

File: GOVERNANCE.md
Comment (bnb) [inline GOVERNANCE.md:116]: I would strongly recommend keeping these documented in a different place. Governance is usually more of a static document, and parts that are going to be changing with frequency are often better kept elsewhere both to make them easier to change and to allow easier historical parsing of a Governance doc. Going through 50 membership changes to find a specific language change is often less than ideal.

File: GOVERNANCE.md
Comment (bnb) [inline GOVERNANCE.md:185]: Caps for consistency with the rest of the bullets in the document.

```suggestion
- Taking no further action
- Issuing a reprimand (private or public)
- Asking for an apology (private or public)
- Permanent ban from the GitHub org and Discord server
- Revoked contributor or moderator status
```

**Review by bnb — COMMENTED** [review]
Hey! I've worked in Node.js, Electron, and the OpenJS Foundation for a number of years on various Governance tasks. I'm excited and hopeful for biome, and wanted to take this opportunity to communicate some of the lessons I've learned over the years in hopes it'll be helpful for y'all, accelerate the project's maturity, and help you avoid some pain.

No worries if you don't take any of the advice or suggestions, but did just want to put them out there for you.

File: GOVERNANCE.md
Comment (Conaclos) [reply GOVERNANCE.md:13]: An odd number of leaders could avoid the issue?

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md:50]: I am on the fence on this one. I am not a big fan about rejection in place, I think I'll remove it. It's very difficult prove a rejection in place. Thank you for raising the issue

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md:5]: > Are there specific criteria for someone to be considered a "Core Contributor" or "Maintainer"? 

A core maintainer should be first a maintainer. Other than that, I don't think there are specific criteria. Do you have anything in mind?

> Are there any conditions under which a Lead might lose their status or privileges?

I would say: inactivity, violation of CoC and step-down. Should we write it somewhere? If so, where?

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md:121]: > Are there any specific criteria or steps to set the project's goals? 

I haven't put much thought into that. Do you have some idea?

> Who is responsible for ensuring the roadmap's goals are being met?

I don't think we should be too strict, for many reasons:

- people work on the project in their own spare time;
- people might become inactive for personal reasons;
- people might not enjoy working on something anymore;

Although, I think people should be held accountable. Should we frame it somewhere? Do you have any suggestion?

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md]: I reworded the sentence. I made it more generic. Let me know if you still have doubts.

**Comment by ematipico** [comment]
> It might be okay to add this later, but it would be good to have an example of what specific contributions should be made in order to be in a certain role. I haven't seen that in many other repositories though.

I added a paragraph: https://github.com/biomejs/biome/blob/chore/governance/GOVERNANCE.md#contributions

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md]: I updated the document, only one lead should step up and make the decision.

@Conaclos that wouldn't work. Being a lead is essentially being a core contributor with more burden (conflict resolution rights, holding sensitive info/assets). We can't force people to join only because we require an odd number.

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md]: I rephrased the sentence. Thank you

**Comment by ematipico** [comment]
> Hey! I've worked in Node.js, Electron, and the OpenJS Foundation for a number of years on various Governance tasks. I'm excited and hopeful for biome, and wanted to take this opportunity to communicate some of the lessons I've learned over the years in hopes it'll be helpful for y'all, accelerate the project's maturity, and help you avoid some pain.
> 
> No worries if you don't take any of the advice or suggestions, but did just want to put them out there for you.

Thank you @bnb ! I reviewed your comments and made the suggested changes. About the "off-boarding" system, what do you recommend? Any word of advice? I rephrased the paragraph, but I am not sure I am on the right track.


@denbezrukov @unvalley since you had questions, I dismissed your approval and requested a review again. I would prefer to have all the questions answered first.

File: GOVERNANCE.md
Comment (denbezrukov) [reply GOVERNANCE.md:145]: > I haven't put much thought into that. Do you have some idea?
I don't have a clear strategy on how to improve it. Let's keep it as it is for now; a roadmap document is fine

**Review by unvalley — APPROVED** [review]
Thank you for update!

**Review by Conaclos — COMMENTED** [review]
I left some comments.

We could also add a diagram to highlight roles and how one is promoted to another. However, this is an enhancement we can do later.

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md:161]: I rephrased the paragraph

File: GOVERNANCE.md
Comment (ematipico) [reply GOVERNANCE.md:65]: I rephrased the paragraph, the voting is a requirement of the role. We can change it in the future once we have more people, and if this become an issue.

**Pull request merged**
